src/main/java/com/github/wovnio/wovnjava/SubdomainUrlLanguagePatternHandler.java

```diff
@@ -1,55 +1,76 @@
 package com.github.wovnio.wovnjava;
 
-import java.util.ArrayList;
 import java.util.regex.Pattern;
 
 class SubdomainUrlLanguagePatternHandler extends UrlLanguagePatternHandler {
     private Lang defaultLang;
-    private ArrayList<Lang> supportedLangs;
+    private LanguageAliases languageAliases;
     private Pattern getLangPattern;
 
-    SubdomainUrlLanguagePatternHandler(Lang defaultLang, ArrayList<Lang> supportedLangs) {
+    SubdomainUrlLanguagePatternHandler(Lang defaultLang, LanguageAliases languageAliases) {
         this.defaultLang = defaultLang;
-        this.supportedLangs = supportedLangs;
+        this.languageAliases = languageAliases;
         this.getLangPattern = this.buildGetLangPattern();
     }
 
     Lang getLang(String url) {
-        Lang lang = this.getLangMatch(url, this.getLangPattern);
-        return (lang != null && this.supportedLangs.contains(lang)) ? lang : null;
+        String languageIdentifier = this.resolvePatternMatch(url, this.getLangPattern);
+        Lang lang = this.languageAliases.getLanguageFromAlias(languageIdentifier);
```

Comment:
Seems like this code doesn't handle if the language identifier isn't an alias? `getLanguageFromAlias` strictly only handles alias -> original

Reply:
`LanguageAliases` class will set up with the standard language code as the "alias" for all supported languages, unless another alias is configured using `langCodeAliases` setting.

It will currently allow only one string to identify a language, such that if "us" is configured as an alias for English, `"en"` will no longer be used to identify it.

Reply:
ah. If all languages are considered their own alias by default then that makes sense.